File: py-project/Solution53-3.py
# -*- coding:utf-8 -*-
class Solution:
    def GetNumberSameAsIndex(self, data):
        length = len(data)
        if data == None or length<=0:
            return -1
        left = 0
        right = length-1
        while left<=right:
            mid = (right+left)//2
            if data[mid] == mid:
                return mid
            elif data[mid] > mid:
                right = mid-1
            else:
                left = mid+1
        return -1


if __name__ == '__main__':
    l = [-3,-1,1,3,5]
    k = 3
    # print(len(l))
    solution1 = Solution()
    a = solution1.GetNumberSameAsIndex(l)
    print(a)

File: c2-project/Solution53-3.cpp
//
// Created by 陈高杰 on 2020/2/17.
//
#include <iostream>
#include <stdlib.h>
#include <string>
#include <vector>
#include <stack>
#include <queue>
#include <math.h>
#include <sstream>
#include <map>
using namespace std;

class Solution {
public:
    int GetNumberSameAsIndex(vector<int> data) {
        int length = data.size();
        if(data.empty() || length <=0)
            return -1;
        int left = 0;
        int right =length-1;
        while(left<=right){
            int mid = (left+right)/2;
            if(data[mid] == mid){
                return mid;
            }
            else if(data[mid] > mid){
                right = mid-1;
            }
            else{
                left = mid+1;
            }
        }
        return -1;
    }
};

//int main(){
////    vector<int> l = {1,2,3,3,3,3,4,5};
//    vector<int> l = {-3,-1,1,3,5};
//    Solution solution;
//    int k = 2;
//    int x = solution.GetNumberSameAsIndex(l);
//    cout<< x<<endl;
//    system("PAUSE");
//    return 0;
//}